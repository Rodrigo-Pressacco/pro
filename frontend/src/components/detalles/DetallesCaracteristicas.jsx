import React from "react";
import "../../styles/detalles/DetallesCaracteristicas.css";

const DetallesCaracteristicas = (props) => {

    return (
        <div className="div-detalles-caracteristicas">
            <div className="div-detalles-h3">
                <h3>Â¿Que ofrece este lugar?</h3>
            </div>
            <div className="div-datosCar">
                {props.caracteristicas.map((e, i) =>
                    <div className="div-datoUnico" key={i}>
                        <i className={e.icono}></i>
                        <p>{e.nombre}</p>
                    </div>)}
            </div>
        </div>
    )
}

export default DetallesCaracteristicas;