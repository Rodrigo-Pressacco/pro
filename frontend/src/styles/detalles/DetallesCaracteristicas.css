.div-detalles-h3{    
    border-bottom: solid var(--main-color) 1px;
}

.div-detalles-h3 h3{
    margin-left: 35px;
    padding: 20px 0;
    font-weight: bold;
    font-size: 25px;
    line-height: 28px;
    color: #3d4161;
}



.div-datosCar{
    display: flex;    
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    margin: 50px 35px;

    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 16px;
    color: #383B58;
}

.div-datoUnico{
    display: flex;    
    align-items: center;
}

.div-datoUnico i{
    margin-right: 20px;
    color: var(--main-color);
    font-family: 'Font Awesome 5 Pro';
}

.columnaNormas{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
}

.tituloNorma {
    margin-top: 30px;
    padding-top: 10px;
    padding-bottom: 30px;
    font-size: 22px;
    color: #4d5172;
}

.columnax3 {
    display: table-cell;  
    margin: 0 30px 20px 30px;
    width: 28%;
    line-height: 40px;
    font-size: 14px;
    color: rgb(104, 97, 97);
}



.textoNormaEspecifico{
    line-height: 24px;
}

.div-detalles-descripcion{
    padding: 35px;
}

.div-detalles-descripcion h3{
    font-weight: bold;
    font-size: 25px;
    line-height: 28px;
    color: #3d4161;
    margin-bottom: 35px;
}

.div-detalles-descripcion p{    
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 35px;
    color: #383B58;
}


@media (max-width: 800px){

    .div-datosCar{
        margin: 30px 35px;
    }

    .div-datoUnico{
        width: 50%;
        height: 60px;
    }


    .columnaNormas{
        justify-content: flex-start;
        flex-direction: row;
    }

    .columnax3{
        width: 42%;
    }


}

@media (max-width: 485px){

    .div-datosCar{
        margin: 15px 15px;
    }

    .div-detalles-h3 h3{
        padding: 15px 0;
        margin-left: 15px;
    }

    .div-datoUnico{
        height: 50px;
    }

    .div-datoUnico i{
        margin-right: 10px;
    }
    
    .columnax3{
        width: 90%;        
    }
    .tituloNorma {
        margin-top: 10px;
    }
    
}