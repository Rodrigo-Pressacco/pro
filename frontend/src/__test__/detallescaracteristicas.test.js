import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import DetallesCaracteristicas from '../components/detalles/DetallesCaracteristicas'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<DetallesCaracteristicas />', () => {
    let component, caracteristicas
    let useEffect = jest.fn(); 
    
    beforeEach(()=> {
        caracteristicas = {
            map : jest.fn()
            
        };
    })

    beforeEach(() => {
        component = shallow(<DetallesCaracteristicas useEffect={useEffect} caracteristicas={caracteristicas} />)
    });

    it("Deberia mostrar <DetallesCaracteristicas/> correctamente", () => {
        expect(component).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta h3", () => {
        let node = component.find("h3")
        expect(node.length).toEqual(1)
    });

})