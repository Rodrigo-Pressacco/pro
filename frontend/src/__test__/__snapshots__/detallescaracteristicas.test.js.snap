// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetallesCaracteristicas /> Deberia mostrar <DetallesCaracteristicas/> correctamente 1`] = `
<div
  className="div-detalles-caracteristicas"
>
  <div
    className="div-detalles-h3"
  >
    <h3>
      Â¿Que ofrece este lugar?
    </h3>
  </div>
  <div
    className="div-datosCar"
  />
</div>
`;
